git-hub

```diff
@@ -294,19 +294,30 @@ def git_dir():
 # The contents of .git/HUB_EDITMSG are returned by the function (after the
 # editor is closed). The text is not filtered at all at this stage (so if the
 # user left the `help_msg`, it will be part of the returned string too.
-def editor(help_msg, msg=None):
+def editor(help_msg, msg=""):
+    PREV_MSG_HEADER="# This is the recovered text as the previous process didn't finish properly.\n"
     prog = git('var', 'GIT_EDITOR')
     fname = os.path.join(git_dir(), 'HUB_EDITMSG')
+
+    if os.path.isfile(fname):
```

Comment:
Yeah, actually I don't think this is quite right. If an user provides a new message you'll just discard it and use the old saved message instead. I don't think this is nice. You should probably leave the `None` as default and here first check `if msg is None and os.path.isfile(fname)` instead. Either that or always prepending, unconditionally, the old message if one is present.
```py
    if os.path.isfile(fname):
        with file(fname, 'r') as f:
            old_msg = PREV_MSG_HEADER + f.read()
    else:
        old_msg = ''
    # If a message was provided, use the old message as a comment
    if msg is not None:
        old_msg = ''.join('# ' + l for l in old_msg.splitlines(True))
    msg = old_msg + (msg or '') + help_msg
```
(untested)